Change 'ekn_id' to 'id' throughout

ptomato opened the pull request:
EKN ID is a term closely connected with eos-knowledge-lib, and we'd like
to avoid that since the intention is to use libdmodel from other places.

(This still leaves ekn:// as the URI scheme, and the EKN_VERSION marker
file, but those probably need to stay so that we don't need to rebuild
the *entire* world.)

https://phabricator.endlessm.com/T22103

ptomato committed:
Change 'ekn_id' to 'id' throughout

EKN ID is a term closely connected with eos-knowledge-lib, and we'd like
to avoid that since the intention is to use libdmodel from other places.

(This still leaves ekn:// as the URI scheme, and the EKN_VERSION marker
file, but those probably need to stay so that we don't need to rebuild
the *entire* world.)

https://phabricator.endlessm.com/T22103

eos-shell-build commented:
Build finished.

ebassi commented on tests/dmodel/testContent.js:
```diff
@@ -22,7 +22,7 @@ describe('Content Object Model', function () {
     it('successfully creates new object from properties', function () {
         contentObject = DModel.Content.new_from_props({
-            ekn_id : 'ekn:text_editors/Emacs',
+            id: 'ekn:text_editors/Emacs',
```
Doesn't this violate the ekn URI syntax?

ptomato replied:
Yes, although there happens to be no validation in this code path. This test was written a long time ago.

ptomato replied:
I added a commit to fix that.

ebassi reviewed (commented):
Looks good

ptomato committed:
tests: Remove nonstandard IDs from test data

Many of these tests were written before the ID format was standardized.
They happen to not use code paths where the ID is validated, so they
pass, but this changes them to be standard.

https://phabricator.endlessm.com/T22103
